### PR TITLE
Add Microsoft Edge Beta v77.0.235.9

### PR DESCRIPTION
**Pull request by coreygo**
After making all changes to the cask:

- [X] `brew cask audit --download microsoft-edge-beta` is error-free.
- [X] `brew cask style --fix microsoft-edge-beta` reports no offenses.
- [X] The commit message includes the cask’s name and version.
- [X] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, if **adding a new cask**:

- [X] Named the cask according to the [token reference].
- [X] `brew cask install microsoft-edge-beta` worked successfully.
- [X] `brew cask uninstall microsoft-edge-beta` worked successfully.
- [X] `brew cask zap microsoft-edge-beta` worked successfully.
- [X] Checked there are no [open pull requests] for the same cask.
- [X] Checked the cask was not [already refused].
- [X] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/Homebrew/homebrew-cask-versions/pulls
[already refused]: https://github.com/Homebrew/homebrew-cask-versions/search?q=is%3Aclosed&type=Issues
[the correct repo]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask
[version-checksum]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/sha256.md#updating-the-sha256

### CODE REVIEW

File: Casks/microsoft-edge-beta.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+                   'com.microsoft.update.agent',
+                 ],
+      pkgutil:   'com.microsoft.package.Microsoft_AutoUpdate.app',
+      trash:     [
+                   '/Library/PrivilegedHelperTools/com.microsoft.autoupdate.helper',
+                   '~/Library/Preferences/com.microsoft.Edge.Beta.plist',
+                 ],
+      rmdir:     [
+                   '/Library/Application Support/Microsoft',
+                   '~/Library/Application Support/Microsoft Edge Beta',
```
Comment (vitorgalvao) [inline Casks/microsoft-edge-beta.rb]: This one belongs in `trash`.

**Comment by vitorgalvao** [comment]
Thank you. Would you mind also giving this `zap` treatment to the other Edge casks?

**Comment by coreygo** [comment]
<details><summary>`brew cask zap microsoft-edge-beta` does work locally and addresses this…</summary>
<p>

```sh
==> brew cask uninstall microsoft-edge-beta ✘
==> Uninstalling Cask microsoft-edge-beta
==> Uninstalling packages:
com.microsoft.Edge.Beta
==> Purging files for version 77.0.235.9 of Cask microsoft-edge-beta
Error: Some packages are still installed, add them to uninstall pkgutil:
com.microsoft.package.Microsoft_AutoUpdate.app
Error: Some launch jobs are still installed, add them to uninstall launchctl:
com.microsoft.autoupdate.helper
com.microsoft.update.agent
Error: Some launch jobs were not unloaded, add them to uninstall launchctl:
com.microsoft.autoupdate.helper
com.microsoft.update.agent
Error: Build failed.
```

</p>
</details>

@vitorgalvao Will go ahead and update the other `microsoft-edge-*` casks. Should I update #7746 or create a new pull request?

**Comment by vitorgalvao** [comment]
> Should I update #7746 or create a new pull request?

Either is fine. Since it doesn’t touch on the same things, a separate PR should add no conflicts.

**Comment by vitorgalvao** [comment]
Thank you for the PR and quick turnaround.

**Pull request merged**